admin/server/assets.go

```diff
@@ -96,3 +196,47 @@ func (s *Server) assetPath(object string) (string, error) {
 	uploadPath.Path = object
 	return uploadPath.String(), nil
 }
+
+func gitToFilesList(gitPath, repo, branch, subpath, token string) ([]drivers.DirEntry, error) {
```

Comment:
Did you consider directly using the download as tar API : https://docs.github.com/en/rest/repos/contents?apiVersion=2022-11-28#download-a-repository-archive-tar

Reply:
It doesnt seem to support subpath.